drivers/t5838/t5838.h

```diff
@@ -0,0 +1,226 @@
+	T5838_AAD_D_FLOOR_75dB = 0x01E0,
+	T5838_AAD_D_FLOOR_80dB = 0x0370,
+	T5838_AAD_D_FLOOR_85dB = 0x062C,
+	T5838_AAD_D_FLOOR_87dB = 0x07BC
+};
+
+/** datasheet claim values between 0x000 and 0x12C are allowed, but only provides us with table of
+ * discrete values defined in this enumerator  */
+enum T5838_aad_d_rel_pulse_min {
+	T5838_AAD_D_REL_PULSE_MIN_0_7ms = 0x0000,
+	T5838_AAD_D_REL_PULSE_MIN_10ms = 0x0064,
+	T5838_AAD_D_REL_PULSE_MIN_19ms = 0x00C8,
+	T5838_AAD_D_REL_PULSE_MIN_29ms = 0x012C
+};
+
+enum T5838_aad_d_abs_pulse_min {
+	T5838_AAD_D_ABS_PULSE_MIN_1_1ms = 0x0000,
+	T5838_AAD_D_ABS_PULSE_MIN_10ms = 0x0064,
+	T5838_AAD_D_ABS_PULSE_MIN_19ms = 0x00C8,
+	T5838_AAD_D_ABS_PULSE_MIN_29ms = 0x012C,
+	T5838_AAD_D_ABS_PULSE_MIN_48ms = 0x01F4,
+	T5838_AAD_D_ABS_PULSE_MIN_95ms = 0x03E8,
+	T5838_AAD_D_ABS_PULSE_MIN_188ms = 0x07D0,
+	T5838_AAD_D_ABS_PULSE_MIN_282ms = 0x0BB8,
+	T5838_AAD_D_ABS_PULSE_MIN_328ms = 0x0DAC
+};
+
+typedef void (*T5838_wake_handler_t)(const struct device *dev);
+
+struct T5838_thconf {
+	/* enum used in all AAD modes */
+	enum T5838_aad_select aad_select;
+
+	/* enums used in AAD A mode */
+	enum T5838_aad_a_lpf aad_a_lpf;
+	enum T5838_aad_a_thr aad_a_thr;
+
+	/* enums used in AAD D modes */
+	enum T5838_aad_d_floor aad_d_floor;
+	enum T5838_aad_d_rel_pulse_min aad_d_rel_pulse_min;
+	enum T5838_aad_d_abs_pulse_min aad_d_abs_pulse_min;
+	enum T5838_aad_d_abs_thr aad_d_abs_thr;
+	enum T5838_aad_d_rel_thr aad_d_rel_thr;
+};
+
+/**
+ * @brief Initialize T5838 device AAD mode
+ *
+ * Function will configure and run T5838 AAD and configure interrupts on wake pin.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param thconf Pointer to the structure containing AAD configuration
+ *
+ * @return int 0 if successful, negative errno code if failure.
+ */
+int t5838_configure_AAD(const struct device *dev, struct T5838_thconf *thconf);
+
+/**
+ * @brief Set AAD wake pin interrupt handler function
+ *
+ * Note that when interrupt gets triggered it will disable further interrupts until
+ * t5838_wake_clear() is called.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param handler Pointer to the handler function
+ */
+void t5838_wake_set_handler(const struct device *dev, T5838_wake_handler_t handler);
+
+/**
+ * @brief Clear AAD wake pin interrupt and reenable AAD interrupts.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ *
+ * @return int 0 if successful, negative errno code if failure.
+ */
+int t5838_wake_clear(const struct device *dev);
+
+/**
+ * @brief Pool AAD wake pin
+ *
+ * note that since T5838 does not have any interrupt flag logic, this is very unreliable way of
+ * reading AAD wake pin. Using interrupts is recommended.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ *
+ * @return int 0 if successful, negative errno code if failure.
+ */
+int t5838_wake_get(const struct device *dev);
```

Comment:
What does this function do? The doxygen comment does not describe the expected behavior, it only provides some internal logic explanation, also what is the function returning?

Reply:
That is mistake on my part. On the second thought I will actually just remove this function, since the way wake pin is driven will lead to unexpected behavior.